crates/sqlexec/src/planner/context_builder.rs

```diff
@@ -111,21 +112,32 @@ impl<'a> PartialContextProvider<'a> {
             match &ent {
                 // References to external databases should always resolve
                 // remotely.
-                ResolvedEntry::NeedsExternalResolution { .. } => {
-                    let owned = reference.to_owned_reference();
-                    let prov = client.dispatch_access(owned).await?;
+                ResolvedEntry::NeedsExternalResolution {
+                    db_ent,
+                    schema,
+                    name,
+                } => {
+                    let prov = client
+                        .dispatch_access(ResolvedTableReference::External {
+                            database: db_ent.meta.name.clone(),
+                            schema: schema.clone().into_owned(),
+                            name: name.clone().into_owned(),
+                        })
+                        .await?;
                     return Ok(Arc::new(prov));
                 }
 
                 // Only resolve native and external tables remotely.
                 //
                 // Temp and system tables should be handled locally.
                 ResolvedEntry::Entry(ent) => {
-                    // TODO: Probably want to get the fully qualified reference.
                     let meta = ent.get_meta();
                     if !(meta.is_temp || meta.builtin) {
-                        let owned = reference.to_owned_reference();
-                        let prov = client.dispatch_access(owned).await?;
+                        let prov = client
+                            .dispatch_access(ResolvedTableReference::Internal {
+                                table_oid: meta.id,
+                            })
+                            .await?;
```

Comment:
We're sending an external table as `ResolvedTableReference::Internal` but not resolving it as an external table in `context/remote.rs`.

Reply:
Will fix.